algorithms/vi-map-helpers/src/vi-map-landmark-quality-evaluation.cc

```diff
@@ -91,4 +144,175 @@ void resetLandmarkQualityToUnknown(
   }
 }
 
+void findTracksOfInferiorDuplicateLandmarkObservations(
+    const vi_map::VIMap& map, const vi_map::Landmark& landmark,
+    vi_map::TrackKeypointMap* tracks_with_keypoints) {
+  // Store landmark observations based on frame and store track ids
+  const vi_map::KeypointIdentifierList& landmark_observations =
+      landmark.getObservations();
+  std::unordered_map<
+      vi_map::VisualFrameIdentifier, vi_map::KeypointIdentifierList>
+      frames_with_observations;
+  vi_map::TrackKeypointMap track_ids_with_observations;
+  std::unordered_map<vi_map::KeypointIdentifier, int>
+      observations_with_track_id;
+  for (const auto& keypoint : landmark_observations) {
+    vi_map::KeypointIdentifierList& frame_observation =
+        frames_with_observations[keypoint.frame_id];
+    frame_observation.push_back(keypoint);
+    const pose_graph::VertexId& vertex_id = keypoint.frame_id.vertex_id;
+    CHECK(vertex_id.isValid());
+    CHECK(map.hasVertex(vertex_id));
+    const vi_map::Vertex& vertex = map.getVertex(vertex_id);
+    CHECK(vertex.hasVisualNFrame());
+    CHECK(vertex.getVisualFrame(keypoint.frame_id.frame_index).isValid());
+    const int track_id = vertex.getVisualFrame(keypoint.frame_id.frame_index)
```

Comment:
You call `vertex.getVisualFrame` twice, here and in the above check, which is just inefficient. Assign it to a variable, check that and then use it further. Also checking that the visual frame is valid is not a very relevant check actually, I would skip it since there's no underlying code there that checks anything meaningful